lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -331,10 +331,7 @@ def __init__(
 
     def _on_refresh(self, event):
         """Trigger the class to update relation data."""
-        if isinstance(event, RelationEvent):
-            relations = [event.relation]
-        else:
-            relations = self._charm.model.relations[self._relation_name]
+        relations = self._charm.model.relations[self._relation_name]
```

Comment:
This would work.
I think the original intention was to reduce unnecessary rel-changed events.
Lists could be unsorted and result in redundant rel-changed events.
In that case we could augment the conditional:
```python
        if isinstance(event, RelationEvent) and event.relation.name == self._relation_name:
            relations = [event.relation]
        else:
            relations = self._charm.model.relations[self._relation_name]
```
On the other hand, I do prefer holistic over deltas.
Wdyt?

Reply:
So, to me it looks like the events of the `cos-agent` relation are not relevant. The way I see it, the `refresh_events` are a list of events **other than** the `cos_agent_relation_changed` (so the second conditional would always be false).

Happy to adjust to this though, if you think that this is better. I just do not understand why a charm would be setting the `cos_agent_relation_changed` event as a refresh event (I understand that this would instead happen on the consumer side only).

Reply:
I don't have a strong opinion on this, for me it's enough that the `cos-agent` relation is updated :)

Reply:
Definitely reads a lot better to me when it says "update the `cos-agent` relations", though